File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 3/1_money_change/change.cpp
#include <iostream>
#include <cmath>
using namespace std;

int change(int amount)
{
    int denominations_10 = floor(amount / 10);
    int denominations_5 = floor((amount - denominations_10 * 10) / 5);
    int denominations_1 = amount - (denominations_10 * 10 + denominations_5 * 5);
    return denominations_1 + denominations_5 + denominations_10;
}

int main()
{
    int amount;
    cin >> amount;
    cout << change(amount);
}

File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 3/1_money_change/change.py
import math


def change(amount):
    denominations_10 = math.floor(amount / 10)
    denominations_5 = math.floor((amount - denominations_10 * 10) / 5)
    denominations_1 = amount - (denominations_10 * 10 + denominations_5 * 5)
    return denominations_1 + denominations_5 + denominations_10


if __name__ == '__main__':
    amount = int(input())
    print(change(amount))
